src/integrations/admin/link-count-tools-integration.php

```diff
@@ -0,0 +1,171 @@
+<?php
+/**
+ * WPSEO plugin file.
+ *
+ * @package Yoast\WP\SEO\Integrations\Admin
+ */
+
+namespace Yoast\WP\SEO\Integrations\Admin;
+
+use WPSEO_Admin_Asset_Manager;
+use Yoast\WP\SEO\Actions\Indexation\Post_Link_Indexing_Action;
+use Yoast\WP\SEO\Actions\Indexation\Term_Link_Indexing_Action;
+use Yoast\WP\SEO\Conditionals\Admin_Conditional;
+use Yoast\WP\SEO\Conditionals\Migrations_Conditional;
+use Yoast\WP\SEO\Conditionals\Yoast_Admin_And_Dashboard_Conditional;
+use Yoast\WP\SEO\Integrations\Integration_Interface;
+use Yoast\WP\SEO\Presenters\Admin\Link_Count_Indexing_List_Item_Presenter;
+use Yoast\WP\SEO\Presenters\Admin\Link_Count_Indexing_Modal_Presenter;
+use Yoast\WP\SEO\Routes\Link_Indexing_Route;
+
+/**
+ * Link_Count_Tools_Integration class
+ */
+class Link_Count_Tools_Integration implements Integration_Interface {
+
+	/**
+	 * @inheritDoc
+	 */
+	public static function get_conditionals() {
+		return [
+			Admin_Conditional::class,
+			Yoast_Admin_And_Dashboard_Conditional::class,
+			Migrations_Conditional::class,
+		];
+	}
+
+	/**
+	 * The post link indexing action.
+	 *
+	 * @var Post_Link_Indexing_Action
+	 */
+	protected $post_link_indexing_action;
+
+	/**
+	 * The term link indexing action.
+	 *
+	 * @var Term_Link_Indexing_Action
+	 */
+	protected $term_link_indexing_action;
+
+	/**
+	 * Represents the admin asset manager.
+	 *
+	 * @var WPSEO_Admin_Asset_Manager
+	 */
+	protected $asset_manager;
+
+	/**
+	 * The total amount of unindexed objects.
+	 *
+	 * @var int
+	 */
+	private $total_unindexed;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Post_Link_Indexing_Action $post_link_indexing_action The post link indexing action.
+	 * @param Term_Link_Indexing_Action $term_link_indexing_action The term link indexing action.
+	 * @param WPSEO_Admin_Asset_Manager $asset_manager             The asset manager.
+	 */
+	public function __construct(
+		Post_Link_Indexing_Action $post_link_indexing_action,
+		Term_Link_Indexing_Action $term_link_indexing_action,
+		WPSEO_Admin_Asset_Manager $asset_manager
+	) {
+		$this->post_link_indexing_action = $post_link_indexing_action;
+		$this->term_link_indexing_action = $term_link_indexing_action;
+		$this->asset_manager             = $asset_manager;
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	public function register_hooks() {
+		\add_action( 'admin_enqueue_scripts', [ $this, 'enqueue_assets' ], 10 );
+		\add_action( 'wpseo_tools_overview_list_items', [ $this, 'render_tools_overview_item' ], 10 );
+	}
+
+	/**
+	 * Enqueues all required assets.
+	 *
+	 * @return void
+	 */
+	public function enqueue_assets() {
+		if ( $this->get_total_unindexed() === 0 ) {
+			return;
+		}
+
+		\add_action( 'admin_footer', [ $this, 'render_modal' ], 20 );
+
+		$this->asset_manager->enqueue_script( 'indexation' );
+		$this->asset_manager->enqueue_style( 'admin-css' );
+
+		$data = [
+			'amount'  => $this->get_total_unindexed(),
+			'ids'     => [
+				'modal'        => '#yoast-link-indexing-modal',
```

Comment:
Are these id's used anywhere else in the PHP?

Reply:
Here and in the presenter. Due to the number of them I felt constants were a bit overkill.

With internal linking finalized it may be worth seeing if we can abstract all these indexing modals into one class so these IDs can get a uniform format which would allow us to abstract them here as well.

Not sure that should be part of this PR though.